wgengine/netstack/netstack.go

```diff
@@ -563,3 +564,7 @@ func startPacketCopy(ctx context.Context, cancel context.CancelFunc, dst net.Pac
 		}
 	}()
 }
+
+func stringifyTEI(tei stack.TransportEndpointID) string {
+	return fmt.Sprintf("%s:%v -> %s:%v", tei.LocalAddress, tei.LocalPort, tei.RemoteAddress, tei.RemotePort)
```

Comment:
I'd suggest `net.JoinHostPort(tei.LocalAddress, tei.LocalPort)` and ditto for remote. That'll do the right thing for IPv6.